core/trino-main/src/main/resources/webapp/src/components/QueryList.jsx

```diff
@@ -509,7 +509,7 @@ export class QueryList extends React.Component {
         }
 
         const filteredQueries = this.filterQueries(this.state.allQueries, newFilters, this.state.errorTypeFilters, this.state.searchString);
-        this.sortAndLimitQueries(filteredQueries, this.state.currentSortType, this.state.currentSortOrder);
+        this.sortAndLimitQueries(filteredQueries, this.state.currentSortType, this.state.currentSortOrder, this.state.maxQueries);
```

Comment:
We should also remove the `!== 0` part of https://github.com/trinodb/trino/blob/30b6b68f6c2f98811684faca8237698d890dddbc/core/trino-main/src/main/resources/webapp/src/components/QueryList.jsx#L274
it becomes redundant now

Reply:
Is it not a performance optimisation for the following two cases - to remove a redundant splice?

https://github.com/trinodb/trino/blob/7989be1420c7901059ee908e91c97d283195da7a/core/trino-main/src/main/resources/webapp/src/components/QueryList.jsx#L362

and

https://github.com/trinodb/trino/blob/7989be1420c7901059ee908e91c97d283195da7a/core/trino-main/src/main/resources/webapp/src/components/QueryList.jsx#L366